browser/tor/tor_profile_service.h

```diff
@@ -0,0 +1,50 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_TOR_TOR_PROFILE_SERVICE_
+#define BRAVE_BROWSER_TOR_TOR_PROFILE_SERVICE_
+
+#include "base/files/file_path.h"
+#include "base/macros.h"
+#include "base/observer_list.h"
+#include "components/keyed_service/core/keyed_service.h"
+#include "net/proxy_resolution/proxy_config.h"
+#include "url/gurl.h"
+
+namespace tor {
+
+struct TorConfig {
+  base::FilePath binary_path;
+  net::ProxyConfig proxy_config;
+};
+
+using TorLaunchCallback =
```

Comment:
Just starting to go through the code, not an issue but just noting it's a bit strange to me this is a callback and not part of `TorLauncherServiceObserver`.